### PR TITLE
port HR-V global

### PR DESCRIPTION
**Pull request by AlexandreSato**
HR-V Global has differences according to your country.

For example the Brazilian does not need an alternative brake signal while the Thai hybrid has different gear signals. So using the Hyundai platform as an example I put the dynamic detection that if it detects canId message from some of these modules the flag to handle is set.

requires: https://github.com/commaai/opendbc/pull/814 (Thai gears messages)

Test routes:
US HR-V: `320098ff6c5e4730|2023-04-14--20-27-00`
Brazilian HR: `147613502316e718|2023-04-10--08-46-37`
Thai HR-V: `1e4baee1aa2687a0|2023-04-14--20-23-45` (https://github.com/commaai/openpilot/pull/27870)

Thai HR-V HDC button: `1e4baee1aa2687a0|2023-04-15--15-53-46--0`
Thai HR-V stock: `1e4baee1aa2687a0|2023-04-15--15-37-29`

**Checklist**
- [x] added entry to CarInfo in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)

### CODE REVIEW

File: selfdrive/car/tests/routes.py [hunk 1]
```diff
@@ -69,7 +69,7 @@
   CarTestRoute("52f3e9ae60c0d886|2021-05-23--15-59-43", HONDA.FIT),
   CarTestRoute("2c4292a5cd10536c|2021-08-19--21-32-15", HONDA.FREED),
   CarTestRoute("03be5f2fd5c508d1|2020-04-19--18-44-15", HONDA.HRV),
-  CarTestRoute("320098ff6c5e4730|2023-04-13--17-47-46", HONDA.HRV_3G),
+  CarTestRoute("147613502316e718|2023-02-24--13-03-54", HONDA.HRV_3G),
```
Comment (sshane) [inline selfdrive/car/tests/routes.py:72]: you can add a new test route

File: selfdrive/car/tests/routes.py [hunk 2]
```diff
@@ -69,7 +69,7 @@
   CarTestRoute("52f3e9ae60c0d886|2021-05-23--15-59-43", HONDA.FIT),
   CarTestRoute("2c4292a5cd10536c|2021-08-19--21-32-15", HONDA.FREED),
   CarTestRoute("03be5f2fd5c508d1|2020-04-19--18-44-15", HONDA.HRV),
-  CarTestRoute("320098ff6c5e4730|2023-04-13--17-47-46", HONDA.HRV_3G),
+  CarTestRoute("147613502316e718|2023-04-10--06-21-49", HONDA.HRV_3G),
```
Comment (sshane) [inline selfdrive/car/tests/routes.py:72]: you can add a new test route instead of replacing

**Comment by sshane** [comment]
Did you figure out how to improve the stopping behavior? We saw plots on Discord of the car not reacting correctly to both the stock system and openpilot's acceleration request when stopping (releases brakes and lurches towards lead).

**Comment by adeebshihadeh** [comment]
Any update on this?

**Comment by AlexandreSato** [comment]
> Any update on this?

We are waiting for a second Brazilian Honda HR-V to confirm the behavior of the PCM, Shane suspects it is a defect in the Kotarou's HR-V. We estimate a test on January 9th, we will use the Harness borrowed from Kotarou in another HR-V (which will have a comma3 only in March)

**Comment by AlexandreSato** [comment]
The creep is only caused by the stock Honda longitudinal (it's a feature known by the owners themselves), the longitudinal control of the openpilot is perfect. PR #30235 can be merged without being under the dashcam flag. As you can see in this https://youtu.be/Q11veROVwF4?t=27m YouTube video that we recorded during the test of the second Brazilian HR-V 2023, openpilot longitudinal control is a tremendous upgrade compared to stock control.

**Comment by sshane** [comment]
We've moved the car interfacing code to our [`opendbc`](https://github.com/commaai/opendbc) repository, which is now the new home for car ports and fingerprints. Please re-open your pull request against [`opendbc`](https://github.com/commaai/opendbc) at your convenience by using this command below. This will transform all changes under `selfdrive/car/` to `opendbc_repo/opendbc/car/`. Make sure you have initialized submodules beforehand and have checked out the latest `opendbc` commit.

```bash
PR_NUMBER=33045
curl -L https://github.com/commaai/openpilot/pull/$PR_NUMBER.patch | sed -e 's/selfdrive\/car/opendbc_repo\/opendbc\/car/g' | git apply -v --reject
```

Simply replace the PR number with your own. Once done, add the files, fix any conflicts, and open a new PR. Alternatively, you may start a new PR from scratch if that is easier for you.

**Pull request closed without merging**